### PR TITLE
[Foundation] Update search form

### PR DESCRIPTION
**Pull request by floriank**
https://community.openproject.org/work_packages/19174

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://floriank-fix-19174-apply-filter-style-to-search.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://floriank-fix-19174-apply-filter-style-to-search.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: app/helpers/search_helper.rb [hunk 1]
```diff
@@ -94,8 +94,7 @@ def project_select_tag
     options << [l(:label_my_projects), 'my_projects'] unless User.current.memberships.empty?
     options << [l(:label_and_its_subprojects, @project.name), 'subprojects'] unless @project.nil? || @project.descendants.active.empty?
     options << [@project.name, 'current_project'] unless @project.nil?
-    label_tag('scope', l(:description_project_scope), class: 'hidden-for-sighted') +
```
Comment (NobodysNightmare) [inline app/helpers/search_helper.rb:97]: Looks like you removed the label without replacement. At least impaired users will notice a difference.

Probably out of scope: To be honest, even non-impaired users might want to know that this select box is for the "Search Scope".

edit: I jsut noticed, that you did not forget that label, but moved it outwards... everythings fine then, except that I think normal users would like it too... still out of scope...

**Comment by NobodysNightmare** [comment]
All in all a great improvement, also over the original search form. If I had to find one thing I don't like it would be that the options for the search query and the scope are not separated in single column mode:

search

Dual columns are fine, though...

edit: Just to be clear, I don't see a blocker here... But if you say a fix is quick and easy, we might want to have it.

**Comment by NobodysNightmare** [comment]
:+1: 

@floriank What you you think about this single-column issue? Fix or ignore for now?

**Comment by floriank** [comment]
I was confused for a moment. I am not sure if there is already something prepared in foundation, i have to check that.

**Comment by floriank** [comment]
@NobodysNightmare I added a modificator to include a border and some space for the checkbox set in single column context

**Comment by floriank** [comment]
Summoning @myabc to check on how much I suck at CSS.

**Comment by NobodysNightmare** [comment]
Someone might say that you added a special case for the last li there. This special case is very special and not generic...
Someone might say thats this is bad.

How about simply applying padding to all `li`s ? This will probably not work everywhere, right?

**Comment by floriank** [comment]
As discussed, i added a modificator, as one cannot know how many filter columns there are and so you cannot really aim for a generic solution.

**Pull request merged**